### PR TITLE
Add breadcrumb to Stores & Sitemap front controllers

### LINKED ISSUE #17131: [EPIC] - Standardize breadcrumbs in Front-office
# Breadcrumbs in the Front-office

## Goal
The goal of this EPIC is to clarify the rule about breadcrumbs in PrestaShop Front-office and to adapt existing/add missing breadcrumbs according to it.

## Why
Several issues have been opened in the previous months about breadcrumbs. 
Some PRs on this topic have been submitted as well.

As of today, there is no clear rule defined about breadcrumbs in the FO. We need to define one and to implement it.

## Related issues

- [x] #29539

### Related issues
- [x] #15440 - No breadcrumb on some important pages (like Stores, Sitemap, New Products, Best Sales, Suppliers, Cart...)
- [x] #16836 - Breadcrumb missing on my account page
- [x] #22416 - SEO-Problem: Breadcrumblist error
- [x] #23530 - Desactivated category still display in breadcrumbs
- [x] #26261

### Specs
- [Specs for breadcrumbs in Front-office ](https://github.com/PrestaShop/prestashop-specs/pull/88/files )

### Implementation 
Missing breadcrumbs to add:
- [ ] TBC

Existing breadcrumbs to remove:
- [ ] Breadcrumbs from CMS pages because they don't have more than 2 depths. 

Existing breadcrumbs to remove when user isn't logged in:
- [ ] /index.php?controller=Identity
- [ ] /index.php?controller=OrderHistory
- [ ] /index.php?controller=CreditSlip
- [ ] /index.php?controller=Addresses

Existing breadcrumbs to modify:
- [ ] /index.php?controller=Search (product page): 
Replace 'Home > Clothes > Women > Hummingbird t-shirt' 
by Home > Search results for hummingbird > Hummingbird printed t-shirt'

### PR DESCRIPTION
**Pull request by nprokopenko202**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Add missing breadcrumbs in Stores and Sitemap pages in Front-office
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #17131.
| How to test? | It partially fix #17131<br><br>So go to page :<br>* /index.php?controller=stores > Now there is a breadcrumb<br>* /index.php?controller=sitemap > Now there is a breadcrumb

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/19195)

### CODE REVIEW

**Comment by Matt75** [comment]
Welcome 202 😉

File: controllers/front/SitemapController.php [hunk 1]
```diff
@@ -186,4 +186,16 @@ protected function getOffersLinks()
...
         return $links;
     }
+
+    public function getBreadcrumbLinks()
+    {
+        $breadcrumb = parent::getBreadcrumbLinks();
+
+        $breadcrumb['links'][] = [
+            'title' => $this->trans('Sitemap', [], 'Shop.Theme.Global'),
+            'url' => $this->context->link->getPageLink('sitemap', true)
```
Comment (Matt75) [inline controllers/front/SitemapController.php]:
```suggestion
            'url' => $this->context->link->getPageLink('sitemap', true),
```

File: controllers/front/StoresController.php [hunk 2]
```diff
@@ -233,4 +233,16 @@ public function getTemplateVarStores()
...
         return $stores;
     }
+
+    public function getBreadcrumbLinks()
+    {
+        $breadcrumb = parent::getBreadcrumbLinks();
+
+        $breadcrumb['links'][] = [
+            'title' => $this->trans('Our stores', [], 'Shop.Theme.Global'),
+            'url' => $this->context->link->getPageLink('stores', true)
```
Comment (Matt75) [inline controllers/front/StoresController.php]:
```suggestion
            'url' => $this->context->link->getPageLink('stores', true),
```

File: controllers/front/StoresController.php [hunk 3]
```diff
@@ -233,4 +233,16 @@ public function getTemplateVarStores()
...
+        ];
+
```
Comment (Matt75) [inline controllers/front/StoresController.php]:
```suggestion

```

**Review by Matt75 — CHANGES_REQUESTED** [review]
Make php-cs-fixer 🟢

File: controllers/front/StoresController.php [hunk 4]
```diff
@@ -233,4 +233,15 @@ public function getTemplateVarStores()
...
+    public function getBreadcrumbLinks()
+    {
+        $breadcrumb = parent::getBreadcrumbLinks();
+
+        $breadcrumb['links'][] = [
+            'title' => $this->trans('Our stores', [], 'Shop.Theme.Global'),
+            'url' => $this->context->link->getPageLink('stores', true),
+        ];
+        return $breadcrumb;
```
Comment (Matt75) [inline controllers/front/StoresController.php]:
```suggestion
        ];

        return $breadcrumb;
```

**Review by Matt75 — CHANGES_REQUESTED** [review]

**Review by Matt75 — APPROVED** [review]

**Comment by nprokopenko202** [comment]
Hello, @Matt75 Thanks a lot!

**Comment by Progi1984** [comment]
@ Thank you for your contribution. :hugs: 

Could you give us a scenario which permits to test the bug before and after this PR ?

Thanks :+1:

**Review by Progi1984 — APPROVED** [review]

**Comment by clotairer** [comment]
> @ Thank you for your contribution. hugs
> 
> Could you give us a scenario which permits to test the bug before and after this PR ?
> 
> Thanks +1

It partially fix #17131

So go to page :
* /index.php?controller=stores > Now there is a breadcrumb
* /index.php?controller=sitemap > Now there is a breadcrumb

**Comment by Progi1984** [comment]
Thnaks @clotaire202  / @nprokopenko202

**Review by PierreRambaud — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Comment by SD1982** [comment]
LGTM Thanks @nprokopenko202

**Comment by Progi1984** [comment]
Thanks @nprokopenko202

**Pull request merged**